code/modules/mob/living/simple_animal/hostile/retaliate/giant_lizard.dm

```diff
@@ -345,15 +488,20 @@
 		stance = HOSTILE_STANCE_IDLE
 
 	//if we're hungry and we don't have already have our eyes on a snack, try eating food if possible
-	if(!food_target && COOLDOWN_FINISHED(src, food_cooldown))
+	if(tameable && !food_target && COOLDOWN_FINISHED(src, food_cooldown))
 		for(var/obj/item/reagent_container/food/snacks/food in view(6, src))
-			if(!is_type_in_list(food, acceptable_foods))
-				continue
-			food_target = food
-			stance = HOSTILE_STANCE_ALERT
-			stop_automated_movement = TRUE
-			MoveTo(food_target)
-			break
+			var/is_meat = FALSE
+			for(var/datum/reagent/nutriment/meat/meat in food.reagents.reagent_list)
+				if(istype(meat))
+					is_meat = TRUE
+					break
```

Comment:
iterating through a list like this already typechecks, so is redundant. but you can also just do this

```suggestion
			var/is_meat = locate(/datum/reagent/nutriment/meat) in food.reagents.reagent_list
```